astroquery/linelists/cdms/core.py

```diff
@@ -303,12 +331,66 @@ def tryfloat(x):
 CDMS = CDMSClass()
 
 
+def parse_letternumber(st):
+    """
+    Parse CDMS's two-letter QNs
+
+    From the CDMS docs:
+    "Exactly two characters are available for each quantum number. Therefore, half
+    integer quanta are rounded up ! In addition, capital letters are used to
+    indicate quantum numbers larger than 99. E. g. A0 is 100, Z9 is 359. Small
+    types are used to signal corresponding negative quantum numbers."
+    """
+    asc = string.ascii_lowercase
+    ASC = string.ascii_uppercase
+    newst = ''.join(['-' + str(asc.index(x)+10) if x in asc else
+                     str(ASC.index(x)+10) if x in ASC else
+                     x for x in st])
+    return int(newst)
+
+
+class Lookuptable(dict):
+
+    def find(self, st, flags):
+        """
+        Search dictionary keys for a regex match to string s
+
+        Parameters
+        ----------
+        s : str
+            String to compile as a regular expression
+            Can be entered non-specific for broader results
+            ('H2O' yields 'H2O' but will also yield 'HCCCH2OD')
+            or as the specific desired regular expression for
+            catered results, for example: ('H20$' yields only 'H2O')
+
+        flags : int
+            Regular expression flags.
+
+        Returns
+        -------
+        The list of values corresponding to the matches
+
+        """
+
+        out = {}
+
+        for kk, vv in self.items():
+            # note that the string-match attempt here differs from the jplspec
+            # implementation
+            match = (st in kk) or re.search(st, str(kk), flags=flags)
+            if match:
+                out[kk] = vv
+
+        return out
+
+
 def build_lookup():
```

Comment:
out of curiosity, what's the reason of this being factored out from the main class?

Reply:
I don't remember.  😬